sockeye/data_io.py

```diff
@@ -436,7 +449,8 @@ def load(self,
             num_pad_target += buck[1] - target_len
 
             sample_index = bucket_sample_index[buck_index]
-            data_source[buck_index][sample_index, :source_len] = source
+            for i, s in enumerate(sources):
```

Comment:
before I forget: do we check somewhere that the sources are indeed token parallel?

Reply:
that is a good point! I don't think so. Will take a look...

Reply:
Actually we didn't. I added a check in DataSetLoader.load and in shard_data

Reply:
cool, thanks!